File: cs/GildedRoseTests/GildedRoseTest.cs
using Xunit;
using System.Collections.Generic;
using GildedRoseKata;
using System.IO;
using System.Runtime.CompilerServices;

namespace GildedRoseTests
{
    public class GildedRoseTest
    {
        [Fact]
        public void defaultItem()
        {
            Item defaultItem = new Item { Name = "default", SellIn = 20, Quality = 10 };
            IList<Item> Items = new List<Item> { defaultItem };
            GildedRose shop = new GildedRose(Items);

            shop.UpdateQuality();
            Assert.Equal(19, defaultItem.SellIn);
            Assert.Equal(9, defaultItem.Quality);
        }

        [Fact]
        public void goldenMaster() {
            IList<Item> items = new List<Item>{
                new Item {Name = "+5 Dexterity Vest", SellIn = 10, Quality = 20},
                new Item {Name = "Aged Brie", SellIn = 2, Quality = 0},
                new Item {Name = "Elixir of the Mongoose", SellIn = 5, Quality = 7},
                new Item {Name = "Sulfuras, Hand of Ragnaros", SellIn = 0, Quality = 80},
                new Item {Name = "Sulfuras, Hand of Ragnaros", SellIn = -1, Quality = 80},
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 5,
                    Quality = 49
                },
				// this conjured item does not work properly yet
				new Item {Name = "Conjured Mana Cake", SellIn = 3, Quality = 6}
            };

            var app = new GildedRose(items);
            IList<string> actualLog = new List<string>();

            for (var i = 0; i < 31; i++)
            {
                actualLog.Add("-------- day " + i + " --------");
                actualLog.Add("name, sellIn, quality");
                for (var j = 0; j < items.Count; j++)
                {
                    actualLog.Add(items[j].Name + ", " + items[j].SellIn + ", " + items[j].Quality);
                }
                app.UpdateQuality();
            }

            string thisPath = GetThisFilePath();
            string thisDirectory = Path.GetDirectoryName(thisPath);


            IList<string> expectedLog = new List<string>();
            string expectedLogPath = thisDirectory + "/../../golden-master/expected-output.txt";
            foreach (string line in File.ReadLines(expectedLogPath)) {
                if(line.Length != 0) {
                    expectedLog.Add(line);
                }
            }

            Assert.Equal(expectedLog, actualLog);

        }

        private static string GetThisFilePath([CallerFilePath] string path = null)
        {
            return path;
        }
    
    }
}

File: java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

import org.junit.jupiter.api.Test;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class GildedRoseTest {

    @Test
    public void simpleItem() {
        Item foo = new Item("foo", 20, 10);
        Item[] items = new Item[] { foo  };
        GildedRose gildedRose = new GildedRose(items);

        gildedRose.updateQuality();

        assertEquals(19,foo.sellIn);
        assertEquals(9, foo.quality);
    }

    @Test
    public void goldenMaster() throws IOException {
        Item[] items = new Item[] {
            new Item("+5 Dexterity Vest", 10, 20), //
            new Item("Aged Brie", 2, 0), //
            new Item("Elixir of the Mongoose", 5, 7), //
            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
            // this conjured item does not work properly yet
            new Item("Conjured Mana Cake", 3, 6) };

        GildedRose app = new GildedRose(items);

        int days = 30;
        StringBuilder actualLog = new StringBuilder();
        for (int i = 0; i <= days; i++) {
            actualLog.append("-------- day " + i + " --------\n");
            actualLog.append("name, sellIn, quality\n");
            for (Item item : items) {
                actualLog.append(item.toString());
                actualLog.append("\n");
            }
            app.updateQuality();
        }

        StringBuilder expectedLog = new StringBuilder();
        String workingDir = System.getProperty("user.dir");
        String expectedPath = workingDir + "/../golden-master/expected-output.txt";
        File expectedFile = new File(expectedPath);
        FileReader reader = new FileReader(expectedFile);
        BufferedReader bufferedReader = new BufferedReader(reader);
        String line;
        while ((line = bufferedReader.readLine()) != null) {
            if(! line.isEmpty()) {
                expectedLog.append(line);
                expectedLog.append("\n");
            }
        }
        assertEquals(expectedLog.toString(), actualLog.toString());
    }


}
